### PR TITLE
Use `CryptoApi.getKeyBackupInfo` instead of deprecated `MatrixClient.getKeyBackupVersion`

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] I have licensed the changes to Element by completing the [Contributor License Agreement (CLA)](https://cla-assistant.io/element-hq/element-web)

Task https://github.com/element-hq/element-web/issues/26922
Blocked by https://github.com/matrix-org/matrix-js-sdk/issues/4534

Use `CryptoApi.getKeyBackupInfo` instead of deprecated `MatrixClient.getKeyBackupVersion`

### CODE REVIEW

File: src/DeviceListener.ts [hunk 1]
```diff
@@ -230,12 +230,14 @@ export default class DeviceListener {
...
+            crypto &&
+            (!this.keyBackupInfo ||
+                !this.keyBackupFetchedAt ||
+                this.keyBackupFetchedAt < now - KEY_BACKUP_POLL_INTERVAL)
         ) {
-            this.keyBackupInfo = await this.client.getKeyBackupVersion();
+            this.keyBackupInfo = await crypto.getKeyBackupInfo();
             this.keyBackupFetchedAt = now;
         }
         return this.keyBackupInfo;
```
Comment (t3chguy) [inline src/DeviceListener.ts:243]: Can we simplify this with a `if (!crypto) return null;`?

File: src/components/structures/MatrixChat.tsx [hunk 2]
```diff
@@ -1638,7 +1638,7 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
             } else {
                 // otherwise check the server to see if there's a new one
                 try {
-                    newVersionInfo = await cli.getKeyBackupVersion();
+                    newVersionInfo = (await cli.getCrypto()?.getKeyBackupInfo()) || null;
```
Comment (t3chguy) [inline src/components/structures/MatrixChat.tsx:1641]:
```suggestion
                    newVersionInfo = (await cli.getCrypto()?.getKeyBackupInfo()) ?? null;
```

File: src/components/views/dialogs/security/RestoreKeyBackupDialog.tsx [hunk 3]
```diff
@@ -258,7 +258,7 @@ export default class RestoreKeyBackupDialog extends React.PureComponent<IProps,
         });
         try {
             const cli = MatrixClientPeg.safeGet();
-            const backupInfo = await cli.getKeyBackupVersion();
+            const backupInfo = (await cli.getCrypto()?.getKeyBackupInfo()) || null;
```
Comment (t3chguy) [inline src/components/views/dialogs/security/RestoreKeyBackupDialog.tsx:261]:
```suggestion
            const backupInfo = (await cli.getCrypto()?.getKeyBackupInfo()) ?? null;
```

File: src/components/views/settings/SecureBackupPanel.tsx [hunk 4]
```diff
@@ -118,7 +118,7 @@ export default class SecureBackupPanel extends React.PureComponent<{}, IState> {
         this.getUpdatedDiagnostics();
         try {
             const cli = MatrixClientPeg.safeGet();
-            const backupInfo = await cli.getKeyBackupVersion();
+            const backupInfo = (await cli.getCrypto()?.getKeyBackupInfo()) || null;
```
Comment (t3chguy) [inline src/components/views/settings/SecureBackupPanel.tsx:121]:
```suggestion
            const backupInfo = (await cli.getCrypto()?.getKeyBackupInfo()) ?? null;
```

File: src/stores/SetupEncryptionStore.ts [hunk 5]
```diff
@@ -125,7 +125,7 @@ export class SetupEncryptionStore extends EventEmitter {
         this.emit("update");
         try {
             const cli = MatrixClientPeg.safeGet();
-            const backupInfo = await cli.getKeyBackupVersion();
+            const backupInfo = (await cli.getCrypto()?.getKeyBackupInfo()) || null;
```
Comment (t3chguy) [inline src/stores/SetupEncryptionStore.ts:128]:
```suggestion
            const backupInfo = (await cli.getCrypto()?.getKeyBackupInfo()) ?? null;
```

**Review by t3chguy — APPROVED** [review]

File: src/components/structures/MatrixChat.tsx (hunk 2 above)
Comment (florianduros) [reply src/components/structures/MatrixChat.tsx:1641]: `cli.getCrypto()?.getKeyBackupInfo()` can return `undefined`, `null` or `KeyBackupInfo`. Why forcing the nullish coalescing ?

Comment (t3chguy) [reply src/components/structures/MatrixChat.tsx:1641]: nullish coalescing works for both `undefined` & `null` and is more strictly speaking accurate here, ultimately they do the same thing for the return types specified, but are more safe as they do not handle `"" | 0 | false` in the same manner so will trip type checks if the return type was to ever be changed to include `string | boolean | number`

Comment (florianduros) [reply src/components/structures/MatrixChat.tsx:1641]: If we want to use it by default instead of `||`, we should specify it in the code style

Comment (t3chguy) [reply src/components/structures/MatrixChat.tsx:1641]: The code style doesn't state to use `||` either - one could infer that one should use `if` statements or ternary everywhere in the absence of such a rule

Comment (florianduros) [reply src/components/structures/MatrixChat.tsx:1641]: I mean nullish coalescing is clearly a complement/improvement of the old/casual `||`.

**Pull request merged**